humanoid_league_team_communication/humanoid_league_team_communication/humanoid_league_team_communication.py

```diff
@@ -1,438 +1,229 @@
         self.node.create_subscription(Strategy, self.topics['strategy_topic'], self.strategy_cb, 1)
         self.node.create_subscription(Float32, self.topics['time_to_ball_topic'], self.time_to_ball_cb, 1)
-        self.node.create_subscription(ObstacleRelativeArray, self.topics['obstacle_topic'], self.obstacle_cb, 1)
+        self.node.create_subscription(RobotArray, self.topics['robots_topic'], self.robots_cb, 1)
         self.node.create_subscription(PoseStamped, self.topics['move_base_goal_topic'], self.move_base_goal_cb, 1)
 
-    def get_connection(self):
-        self.logger.info(f"Binding to port {self.receive_port}")
-        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
-        sock.bind(('0.0.0.0', self.receive_port))
-        return sock
-
-    def close_connection(self):
-        if self.socket:
-            self.socket.close()
-            self.logger.info("Connection closed.")
-
-    def receive_msg(self):
-        return self.socket.recv(1024)
-
-    def gamestate_cb(self, msg):
+    def gamestate_cb(self, msg: GameState):
         self.gamestate = msg
 
-    def pose_cb(self, msg):
+    def pose_cb(self, msg: PoseWithCovarianceStamped):
         self.pose = msg
 
-    def cmd_vel_cb(self, msg):
+    def cmd_vel_cb(self, msg: Twist):
         self.cmd_vel = msg
-        self.cmd_vel_time = self.node.get_clock().now()
+        self.cmd_vel_time = self.get_current_time().to_msg()
 
-    def strategy_cb(self, msg):
+    def strategy_cb(self, msg: Strategy):
         self.strategy = msg
-        self.strategy_time = self.node.get_clock().now()
+        self.strategy_time = self.get_current_time().to_msg()
 
-    def time_to_ball_cb(self, msg):
+    def time_to_ball_cb(self, msg: float):
         self.time_to_ball = msg.data
-        self.time_to_ball_time = self.node.get_clock().now()
+        self.time_to_ball_time = self.get_current_time().to_msg()
 
-    def move_base_goal_cb(self, msg):
+    def move_base_goal_cb(self, msg: PoseStamped):
         self.move_base_goal = msg
 
-    def obstacle_cb(self, msg):
-        self.obstacles = ObstacleRelativeArray(header=msg.header)
-        self.obstacles.header.frame_id = self.map_frame
-        for obstacle in msg.obstacles:
-            # Transform to map
-            obstacle_pose = PoseStamped(msg.header, obstacle.pose.pose.pose)
+    def robots_cb(self, msg: RobotArray):
+
+        def transform_to_map(robot_relative: Robot):
+            robot_pose = PoseStamped(header=msg.header, pose=robot_relative.bb.center)
             try:
-                obstacle_map = self.tf_buffer.transform(obstacle_pose,
-                                                        self.map_frame,
-                                                        timeout=Duration(nanoseconds=0.3e9))
-                obstacle.pose.pose.pose = obstacle_map.pose
-                self.obstacles.obstacles.append(obstacle)
+                robot_map = self.tf_transform(robot_pose)
```

Comment:
But isn't the target configurable and thus dynamic? Or what do you mean

Reply:
I mean that the variable is called `robots_map` which includes that the robots are in the `map` frame. But the function is called `tf_transform` without an indication to which frame it transforms. So it could be a bit confusing. I would prefer a name like `self.to_map_frame(robot_pose)`. And the name of the map frame might be configurable, but this is manly the case to allow prefixing in a multi robot scenario.